SubscribeIT/ngDesk
# 99 migrate csv import

#### Reference to issue (Required)

Closes #99 

#### Checklist (Required)


#### Description (Required)

The entries will be imported via files with csv and excel formats.

Format and sample payload
[csvRules.txt](https://github.com/SubscribeIT/ngDesk/files/7282792/csvRules.txt)
[tickets.csv](https://github.com/SubscribeIT/ngDesk/files/7282794/tickets.csv)
[allDataTypes.csv](https://github.com/SubscribeIT/ngDesk/files/7282798/allDataTypes.csv)

#### Manual Test Preformed (Required)
List the manual tests that were preformed and the steps to preform them  
**There must be multiple test cases provided, each test case should include the exact and detailed steps to replicate the case**
**I have included 2, test cases here but there will be many time when more cases are required**

1. Goto dev1.ngdesk.com -> company settings -> Import module data from file.
2. Select import, choose module and upload the file.
3. Select the csv headers in the dropdown, for its respective field.
4. After clicking import, the import job takes place.
5. If the status is failed, the entry isn't created.
6. If the status is completed, the entry would be created, but if there are any errors, it will be logged and can be viewed.
7. The line which has errors will be skipped, other entries will be created.
8. Screenshots:
![Screenshot from 2021-10-05 10-25-02](https://user-images.githubusercontent.com/89620702/135962466-23d24da1-c7a5-45b6-bef6-4b5ee11c2c64.png)
![Screenshot from 2021-10-05 10-25-05](https://user-images.githubusercontent.com/89620702/135962471-3fd8b3a4-f72a-4e38-bcb9-07a65d1d2200.png)
![Screenshot from 2021-10-05 10-25-21](https://user-images.githubusercontent.com/89620702/135962473-ad11dc68-ceb4-4572-81bd-b9d5a3e659d1.png)

#### Test Case 1
**Name: Existing module**
1. Importing ticket entries.
2. Select the csv headers with respect to field name.
3. Once its been imported and status is set to completed, the entry will be created.
4. Screenshots:
![Screenshot from 2021-10-05 10-23-44](https://user-images.githubusercontent.com/89620702/135962352-9ff6ff07-ecac-48e7-952c-8359a94c115f.png)
![Screenshot from 2021-10-05 10-23-47](https://user-images.githubusercontent.com/89620702/135962357-0de5ae37-8468-405a-9a41-2b0be93ae941.png)
![Screenshot from 2021-10-05 10-23-58](https://user-images.githubusercontent.com/89620702/135962360-307a1db7-94ae-4567-8015-510da343fe57.png)

#### Test Case 2
**Name: Custom module**
1. Importing ticket entries.
2. Select the csv headers with respect to field name.
3. Once its been imported and status is set to completed, the entry will be created.
4. Screenshots:
![Screenshot from 2021-10-05 10-27-02](https://user-images.githubusercontent.com/89620702/135962607-6c86f905-f1f8-4c61-b8fc-b9702742ad10.png)
![Screenshot from 2021-10-05 10-27-06](https://user-images.githubusercontent.com/89620702/135962614-f69c085f-25e4-4976-9923-36b741a26558.png)
![Screenshot from 2021-10-05 10-27-16](https://user-images.githubusercontent.com/89620702/135962617-971622b8-0f33-4a74-a3ba-17c679af9770.png)

#### List of General Components affected (Required)

#### Types of changes (Required)
